fdbserver/VersionedBTree.actor.cpp

```diff
@@ -7702,11 +7702,16 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 		        : SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_WINDOW_BYTES;
 
 		EncodingType encodingType = EncodingType::XXHash64;
-
-		// Deterministically enable encryption based on uid
-		if (g_network->isSimulated() && logID.hash() % 2 == 0) {
-			encodingType = EncodingType::XOREncryption;
-			m_keyProvider = std::make_shared<XOREncryptionKeyProvider>(filename);
+		// TODO(yiwu): We should enable encryption only when tenant is in required mode, but currently when existing
+		// storage engine instance is reopen on worker server start, the tenant mode is unknown.
+		if (SERVER_KNOBS->ENABLE_STORAGE_SERVER_ENCRYPTION) {
+			encodingType = EncodingType::AESEncryptionV1;
+			ASSERT(encryptionKeyProvider.isValid());
+			m_keyProvider = encryptionKeyProvider;
+		} else if (g_network->isSimulated() && logID.hash() % 2 == 0) {
```

Comment:
any reason we want to write encrypted data when KNOB doesn't support it?

Reply:
Ahh this seems like the default encryption mode. 
I think it is better to have KNOB control encryption, but then the mode needs to choose what encryption scheme is applicable? To me `else` seems bit weird given we directed the cluster NOT to do any encryption?

Reply:
I think I shall remove it from the normal code path, and test the XOREncryption (which is a test only fake encryption mode) in unit test only.

Reply:
I would say there is value in retaining the functionality, it some ways Redwood supports two encryption modes:
1. XOR
2. AES 256 CTR

I think an easier way to incorporate will be extend 

```
typedef enum {
	ENCRYPT_CIPHER_MODE_NONE = 0,
	ENCRYPT_CIPHER_MODE_AES_256_CTR = 1,
	ENCRYPT_CIPHER_MODE_LAST = 2
} EncryptCipherMode;
```

and may be add `ENCRYPT_CIPHER_XOR_MODE`; for db-config change we plan to introduce `encryption_data_at_rest_mode` (std::string), so can enable this across other processes too in future.
Thoughts?

Reply:
The XOR encryption type is a fake encryption type only for testing. No one should be using it, other than testing. Making the db config support it is risky.

Reply:
Do we plan to remove XOR encryption OR remove the code from this particular codepath?

Reply:
I've removed it from this particular codepath.